db/inmemory_test.go

```diff
@@ -0,0 +1,54 @@
+package db
+
+import "testing"
+
+func TestBasic(t *testing.T) {
+	im := NewInMemoryVolumeDatabase()
+
+	err := im.Connect()
+	if err != nil {
+		t.Fatal("Connect returned an error! " + err.Error())
+	}
+
+	list, err := im.List()
+	if err != nil {
+		t.Fatal("List returned an error! " + err.Error())
```

Comment:
List in this function is verifying that the action we expected to happen did. The first one checks initial state, the test that we are mutating correctly. I feel like this is the intended use of fatal. I see error used when there are table/list driven tests where one error call does not depend on the previous error check.